Jessy/buy_me_a_laptop_challenge.py

```diff
@@ -0,0 +1,113 @@
+        },
+        {
+            "id": "2",
+            "productName": "Dell XPS",
+            "url": "https://www.dell.com/en-us/shop/dell-laptops/sr/laptops/xps-laptops?~ck=bt",
+            "types": [
+                {
+                    "id": "3",
+                    "screen_size": "13-inch",
+                    "cpu": ["11th Generation Intel Core i3-1115G4 Processor", "11th Generation Intel Core i5-1135G7 Processor"],
+                    "ram": ["8GB"],
+                    "storage": ["256GB SSD","512 GB SSD", "1 TB SSD"],
+                    "colors": ["Platinum silver exterior, black interior", "Platinum silver exterior, black interior"],
+                    "price": [999, 1099, 1149, 1249]
+                },
+                {
+                    "id": "4",
+                    "screen_size": "15-inch",
+                    "cpu": ["10th Generation Intel Core i5-10300H"],
+                    "ram": ["8GB", "16GB", "32GB", "64GB"],
+                    "storage": ["256 GB SSD", "512 GB SSD", "1 TB SSD", "2 TB SSD"],
+                    "colors": ["Frost White with White Palmrest", "Frost White with White Palmrest"],
+                    "price": [1199, 1299, 1399, 1749, 1999, 2299]
+                }
+            ],
+            "description": "The Dell XPS is an absolutely brilliant laptop. The 2020  version rocks an 11th-generation Intel Core i3, i5 or i7 processor and a bezel-less ‘Infinity Edge’ display. This Dell XPS continues to be the most popular Windows laptop in the world. What’s more, there’s a wide range of customization options, so you can really make the Dell XPS the best laptop for your needs. "
+        }
+]
+
+# TODO: Write code to print out the MacBook Pro url 
+print(laptops[0]['url'])
+
+# TODO: Write code to print all possible prices of the 16-inch MacBook Pro.
+print(laptops[0]['types'][1]['price'])
+
+# TODO: Write code to print all the color options for Dell XPS 13.
+print(laptops[1]['types'][0]['price'])
+
+# TODO: Write code to print the description of Dell XPS laptop.
+print(laptops[1]['description'])
+
+
+print("Question 2: Out of Stock laptops")
+print("Suppose that the 13-inch MacBook Pro in space gray color is sold out. Also, the same laptop with 1 TB storage is out of stock as well. Update the list of dictionaries such that these options are removed. Print the updated dictionary.")
+
+# TODO: Update the laptops dictionary.
+laptops[0]['types'][0]['colors'].pop(0)
+print(laptops[0]['types'][0]['colors'])
+
+laptops[0]['types'][1]['storage'].pop(1)
+print(laptops[0]['types'][1]['storage'])
+# TODO: Print the new dictionary.
+# I printed then in L98 and L101 along with the updated dictionaries
+
+print("Question 3: listing all the prices")
+print("Time to look at the range of prices. Print out all possible computer prices")
+
+# TODO: print out all possible prices for the laptops
+print(laptops[0]['types'][0]['price'])
```

Comment:
clean and simple way to answer the question. Sometime soon lets revisit this, with the `loops` concept